### PR TITLE
add docker shim for running openapi-linter in circleci

### PR DESCRIPTION
**Pull request by ghost**
Closes https://github.com/lightspeed-hospitality/api-platform-squad/issues/107

https://app.circleci.com/pipelines/github/lightspeed-hospitality/OpenTable-Integration/951/workflows/8bc5bde2-84fa-4a00-a64f-bbb5c73477d8/jobs/3426?invite=true#step-112-21

### CODE REVIEW

File: ls_pre_commit_hooks/openapi-linter.sh [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+run_circle_ci() {
+  linter="$1"
+  docker_image="$2"
+  container="$linter-linter-volume"
+
+  shift 2
+
+  if container_exists $container; then
+    docker container rm -f $container
+  fi
```
Comment (ls-samer-abuthaher) [inline ls_pre_commit_hooks/openapi-linter.sh:24]:
```suggestion
  if container_exists "${container}"; then
    docker container rm -f "${container}"
  fi
```

File: ls_pre_commit_hooks/openapi-linter.sh [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+  # create a dummy container which will hold a volume
+  docker create -v /src --name $container busybox /bin/true
+
+  # copy sources into this volume
+  docker cp ${PWD} $container:/src
+
+  # run linter
+  docker run --volumes-from $container --workdir /src $docker_image $linter "$@"
```
Comment (ls-samer-abuthaher) [inline ls_pre_commit_hooks/openapi-linter.sh:33]:
```suggestion
  docker create -v /src --name $container busybox /bin/true

  # copy sources into this volume
  docker cp "${PWD}" "${container}:/src"

  # run linter
  docker run --volumes-from "${container}" --workdir /src "${docker_image}" "${linter}" "$@"
```

File: ls_pre_commit_hooks/openapi-linter.sh [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+run_local() {
+  linter="$1"
+  docker_image="$2"
+
+  shift 2
+
+  docker run -v $PWD:/src:rw,Z --workdir /src $docker_image $linter "$@"
```
Comment (ls-samer-abuthaher) [inline ls_pre_commit_hooks/openapi-linter.sh:42]:
```suggestion
  docker run -v "${PWD}:/src:rw,Z" --workdir /src "${docker_image}" "${linter}" "$@"
```

File: ls_pre_commit_hooks/openapi-linter.sh [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+run() {
+  linter="$1"
+  docker_image="$OPENAPI_LINTER_IMAGE:$OPENAPI_LINTER_VERSION"
```
Comment (ls-samer-abuthaher) [inline ls_pre_commit_hooks/openapi-linter.sh:47]:
```suggestion
  docker_image="${OPENAPI_LINTER_IMAGE}:${OPENAPI_LINTER_VERSION}"
```

File: ls_pre_commit_hooks/openapi-linter.sh [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+  shift
+
+  if [[ -n $CIRCLECI ]]; then
+    echo "Running ${linter} with CircleCI docker volume workaround."
+    run_circle_ci $linter $docker_image "$@"
+  else
+    run_local $linter $docker_image "$@"
```
Comment (ls-samer-abuthaher) [inline ls_pre_commit_hooks/openapi-linter.sh:55]:
```suggestion
  if [[ -n "${CIRCLECI}" ]]; then
    echo "Running ${linter} with CircleCI docker volume workaround."
    run_circle_ci "${linter}" "${docker_image}" "$@"
  else
    run_local "${linter}" "${docker_image}" "$@"
```

File: ls_pre_commit_hooks/openapi-linter.sh [hunk 6]
```diff
@@ -0,0 +1,59 @@
+#!/bin/sh
+set -e
+
+OPENAPI_LINTER_VERSION="${OPENAPI_LINTER_VERSION:-21-73a3f80}"
+OPENAPI_LINTER_IMAGE="809245501444.dkr.ecr.us-east-1.amazonaws.com/release/internal/image/openapi-linter"
+
+container_exists() {
+  if [ $(docker ps -a -f "name=$1" | wc -l) -eq 2 ]; then
```
Comment (ls-samer-abuthaher) [inline ls_pre_commit_hooks/openapi-linter.sh:8]:
```suggestion
  if [ "$(docker ps -a -f "name=$1" | wc -l)" -eq 2 ]; then
```

File: ls_pre_commit_hooks/openapi-linter.sh [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    return 0
+  else
+    return 1
+  fi
+}
+
+run_circle_ci() {
+  linter="$1"
+  docker_image="$2"
+  container="$linter-linter-volume"
```
Comment (ls-samer-abuthaher) [inline ls_pre_commit_hooks/openapi-linter.sh:18]:
```suggestion
  container="${linter}-linter-volume"
```

**Review by ls-jad-elkik — COMMENTED** [review]
I would recommend you run `shfmt` and `shellcheck` on the bash script

**Review by ls-jad-elkik — COMMENTED** [review]
Why `\t` instead of 4 (or 2) spaces?

**Review by ls-jad-elkik — COMMENTED** [review]
You can add those pre-commit hooks if you want:

https://github.com/lightspeed-hospitality/elasticsearch-init/blob/1cbf3d9e91a0c315c1a58f20e2dc540a55f5ee96/.pre-commit-config.yaml#L7-L10

File: ls_pre_commit_hooks/openapi-linter.sh [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+container_exists() {
+	if [ "$(docker ps -a -f name="$1" | wc -l)" -eq 2 ]; then
+		return 0
+	else
+		return 1
+	fi
+}
+
+run_circle_ci() {
```
Comment (ls-jad-elkik) [inline ls_pre_commit_hooks/openapi-linter.sh:15]: Can you add a comment with why we do this and a link to wherever you took this from? A forum post or some stackoverflow answer talking about this, etc.

**Review by ls-jad-elkik — APPROVED** [review]
LGTM if you managed to test it properly and it works 👍

**Pull request merged**